### DISCUSSION TITLE
OTA flash updates

### DISCUSSION THREAD

**Discussion by DanMan32** [PlummersSoftwareLLC/NightDriverStrip#114] [Q&A]
I've read many mention OTA updates, but how does that work with this project and PlatformIO?
I was able to implement OTA for another project of mine last summer which utilized Autoconnect library.   It's main purpose was to provide a means to configure WiFi but as two side benefits provided a JSON web framework and OTA updates by uploading the compiled BIN file using the Autoconnect website with the BIN files generated by the ArduinoIDE platform.
It also provided a means to upload config files to the SPIFF which I had it update configuration parameters for my project instead of having to recompile and upload new code just to change a temperature threshold.
I'm not promoting using the AutoConnect, I encountered some frustration with quirks it had managing WiFi connections, but there's lots of ideas that can be gleaned from it.

**Comment by jurjanbrust** [comment]
Add the following to platformio.ini: 
upload_protocol = espota
upload_port   = 192.168.1.126

Change the port number to your ESP32 device
I commented out 'upload_port' but not sure if that is needed.

And also set the ENABLE_OTA to 1 in globals.h

**Reply by DanMan32** [reply]
Thanks, I'll need to check it out.
I also happened to find the answer on the PlatformIO community forum about the same time you answered:
https://community.platformio.org/t/esp32-ota-using-platformio/15057
